File: codegen/results/v002/ollama/qwen3.8_27b_think/20260827_174503/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    {
        description: "quantity 9 (below 10, no discount)",
        price: 10,
        quantity: 9,
        expected: 10 * 9, // 90
    },
    {
        description: "quantity 10 (boundary, 10% discount)",
        price: 10,
        quantity: 10,
        expected: 10 * 10 * 0.9, // 90
    },
    {
        description: "quantity 49 (just below 50, 10% discount)",
        price: 10,
        quantity: 49,
        expected: 10 * 49 * 0.9, // 441
    },
    {
        description: "quantity 50 (boundary, 25% discount)",
        price: 10,
        quantity: 50,
        expected: 10 * 50 * 0.75, // 375
    },
    {
        description: "quantity 1, price 5 (simple case)",
        price: 5,
        quantity: 1,
        expected: 5 * 1, // 5
    },
    {
        description: "quantity 0 (zero quantity)",
        price: 10,
        quantity: 0,
        expected: 10 * 0, // 0
    },
];

let passed = 0;
let failed = 0;

for (const tc of testCases) {
    const got = calculateDiscount(tc.price, tc.quantity);
    if (Math.abs(got - tc.expected) < 1e-9) {
        console.log(`PASS: ${tc.description}`);
        passed++;
    } else {
        console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${got})`);
        failed++;
    }
}

console.log(`\n${passed} passed, ${failed} failed, ${testCases.length} total`);

File: codegen/results/v002/ollama/qwen3.8_27b_think/20260827_174503/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	epsilon := 1e-9

	tests := []struct {
		description string
		price       float64
		quantity    int
		expected    float64
	}{
		{"quantity 9 (below 10, no discount)", 10.0, 9, 10.0 * 9},
		{"quantity 10 (boundary, 10% discount)", 10.0, 10, 10.0 * 10 * 0.9},
		{"quantity 49 (below 50, 10% discount)", 10.0, 49, 10.0 * 49 * 0.9},
		{"quantity 50 (boundary, 25% discount)", 10.0, 50, 10.0 * 50 * 0.75},
		{"quantity 100 (well above 50, 25% discount)", 5.0, 100, 5.0 * 100 * 0.75},
	}

	for _, tc := range tests {
		got := calculateDiscount(tc.price, tc.quantity)
		if math.Abs(got-tc.expected) < epsilon {
			fmt.Printf("PASS: %s\n", tc.description)
		} else {
			fmt.Printf("FAIL: %s (expected: %v, got: %v)\n", tc.description, tc.expected, got)
		}
	}
}